### PR TITLE
Productionalize `FivetranSensorAsync`

### PR DESCRIPTION
**Pull request by phanikumv**
This PR will 

- Fix all existing bugs in  `FivetranSensorAsync`
- Add unit tests to the hook, sensor and trigger
- Run tests against multiple python versions
- Use `setup.cfg` to mention the package metadata
- Optimize project structure  
- Use `pyproject.toml` to specify build requirements
- Add github worflows to automatically run tests when PR is created

TODO: Auto deploy to PyPi

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+`FivetranSensorAsync` monitors a Fivetran sync job for completion. 
+Monitoring with `FivetranSensorAsync` allows you to trigger downstream processes only when the Fivetran sync jobs have completed, ensuring data consistency. 
+
+
+
+You can use multiple instances of `FivetranSensorAsync` to monitor multiple Fivetran connectors.
+
+If used in this way, 
+
+`FivetranSensorAsync` requires that you specify the `connector_id` of the sync job to start. You can find `connector_id` in the Settings page of the connector you configured in the [Fivetran dashboard](https://fivetran.com/dashboard/connectors).
```
Comment (kaxil) [inline README.md:43]:
```suggestion
If used in this way,  `FivetranSensorAsync` requires that you specify the `connector_id` of the sync job to start. You can find `connector_id` in the Settings page of the connector you configured in the [Fivetran dashboard](https://fivetran.com/dashboard/connectors).
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Import into your DAG via:
+```
+from fivetran_provider_async.sensors.fivetran import FivetranSensorAsync
+```
```
Comment (kaxil) [inline README.md:48]:
```suggestion
```python
from fivetran_provider_async.sensors.fivetran import FivetranSensorAsync
```
```

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+project_urls =
+    Source Code=https://github.com/astronomer/airflow-provider-fivetran-async
+    Changelog=https://github.com/astronomer/airflow-provider-fivetran-async/blob/main/CHANGELOG.md
+    Bug Tracker = https://github.com/astronomer/airflow-provider-fivetran-async/issues
+
+[options]
+python_requires = >=3.6
```
Comment (kaxil) [inline setup.cfg:28]:
```suggestion
python_requires = >=3.7
```

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+license_files = LICENSE.txt
+classifiers =
+    Development Status :: 5 - Production/Stable
+    Environment :: Web Environment
+    Framework :: Apache Airflow
+    Framework :: Apache Airflow :: Provider
+    Intended Audience :: Developers
+    License :: OSI Approved :: Apache Software License
+    Operating System :: OS Independent
+    Programming Language :: Python :: 3.8
```
Comment (kaxil) [inline setup.cfg:19]:
```suggestion
    Programming Language :: Python :: 3.7
    Programming Language :: Python :: 3.8
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  Run-Unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python: [ 3.8, 3.9, '3.10' ]
```
Comment (kaxil) [inline .github/workflows/ci.yml:19]:
```suggestion
        python: [ 3.7, 3.8, 3.9, '3.10' ]
```

Comment (phanikumv) [reply .github/workflows/ci.yml]: @kaxil The tests dont run on 3.7, so we shouldnt add it as per my understanding

**Pull request merged**